File: HR.R
# 4
getOption("repos")
library(readr)
# Importing the Data
hr <- read.csv('C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\HR_DT.csv')
# Performing EDA
str(hr)
table(hr$monthly.income.of.employee)
hr$monthly.income.of.employee <- as.factor(hr$monthly.income.of.employee)
hr$no.of.Years.of.Experience.of.employee <- as.factor(hr$no.of.Years.of.Experience.of.employee)
# Shuffle the data
hrrand <- hr[order(runif(196)), ]
str(hrrand)
# Splitting the data into train and test
train <- hrrand[1:98, ]
test  <- hrrand[99:196, ]
# Check the proportion of class variable
prop.table(table(hrrand$monthly.income.of.employee))
prop.table(table(train$monthly.income.of.employee))
prop.table(table(test$monthly.income.of.employee))
# Training a model on the data
library(C50)
model <- C5.0(train[, -3], train$monthly.income.of.employee)
windows()
plot(model) 
# Display detailed information about the tree
summary(model)
# Evaluating model performance
# Test data accuracy
testres <- predict(model, test)
testacc <- mean(test$monthly.income.of.employee == testres)
testacc
# Cross tabulation of predicted versus actual classes
library(gmodels)
CrossTable(test$monthly.income.of.employee, testres, dnn = c('actual default', 'predicted default'))
# On Training Dataset
trainres <- predict(model, train)
trainacc <- mean(train$monthly.income.of.employee == trainres)
trainacc
table(train$monthly.income.of.employee, trainres)

File: HR.py
# 4
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder
# Importing the dataset
hr = pd.read_csv("C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\HR_DT.csv")
# Checking for null values
hr.isnull().sum()
hr.columns
# Dropping passed columns 
hr.drop(["Position of the employee"], axis = 1, inplace = True) 
# display 
hr 
# Converting into binary
lb = LabelEncoder()
hr["no of Years of Experience of employee"] = lb.fit_transform(hr["no of Years of Experience of employee"])
hr[" monthly income of employee"] = lb.fit_transform(hr[" monthly income of employee"])
hr[' monthly income of employee'].unique()
hr[' monthly income of employee'].value_counts()
colnames = list(hr.columns)
predictors = colnames[:2]
target = colnames[1]
# Splitting data into training and testing data set
from sklearn.model_selection import train_test_split
train, test = train_test_split(hr, test_size = 0.3)
from sklearn.tree import DecisionTreeClassifier as DT
model = DT(criterion = 'entropy')
model.fit(train[predictors], train[target])
# Prediction on Test Data
preds = model.predict(test[predictors])
pd.crosstab(test[target], preds, rownames=['Actual'], colnames=['Predictions'])
np.mean(preds == test[target]) # Test Data Accuracy 
# Prediction on Train Data
preds = model.predict(train[predictors])
pd.crosstab(train[target], preds, rownames = ['Actual'], colnames = ['Predictions'])
np.mean(preds == train[target]) # Train Data Accuracy
